tsl/test/sql/feature_flags.sql

```diff
@@ -0,0 +1,93 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+\c :TEST_DBNAME :ROLE_SUPERUSER
+
+-- hypertable creation
+SHOW timescaledb.enable_hypertable_create;
+SET timescaledb.enable_hypertable_create TO off;
+
+CREATE TABLE test(time timestamptz, device int);
+\set ON_ERROR_STOP 0
+SELECT * FROM create_hypertable('test', 'time');
+\set ON_ERROR_STOP 1
+
+SET timescaledb.enable_hypertable_create TO on;
+SELECT * FROM create_hypertable('test', 'time');
+
+-- hypertable compression
+SHOW timescaledb.enable_hypertable_compression;
+SET timescaledb.enable_hypertable_compression TO off;
+
+INSERT INTO test SELECT t, 0
+FROM generate_series('2018-03-02 1:00'::TIMESTAMPTZ, '2018-03-04 1:00', '1 hour') t;
+SELECT * FROM show_chunks('test');
+
+-- compress_chunk
+\set ON_ERROR_STOP 0
+SELECT compress_chunk('_timescaledb_internal._hyper_1_1_chunk');
+\set ON_ERROR_STOP 1
+
+-- ensure compression cannot be enabled
+\set ON_ERROR_STOP 0
+ALTER TABLE test
+SET (timescaledb.compress,
+     timescaledb.compress_orderby = 'time',
+     timescaledb.compress_segmentby = 'device');
+\set ON_ERROR_STOP 1
+
+-- cagg creation
+SHOW timescaledb.enable_cagg_create;
+SET timescaledb.enable_cagg_create TO off;
+
+\set ON_ERROR_STOP 0
+
+CREATE MATERIALIZED VIEW contagg
+WITH (timescaledb.continuous) AS
+SELECT
+  time_bucket('1 hour', time) AS hour,
+  device
+FROM
+  test
+GROUP BY hour, device;
+
+\set ON_ERROR_STOP 1
+
+SET timescaledb.enable_cagg_create TO on;
+
+CREATE MATERIALIZED VIEW contagg
+WITH (timescaledb.continuous) AS
+SELECT
+  time_bucket('1 hour', time) AS hour,
+  device
+FROM
+  test
+GROUP BY hour, device;
+
```

Comment:
Can you add a check that running refresh with the flag disabled prints a proper error message?

Reply:
Sure, but that assumes that cagg was created before we disable it in the feature flags. Just thinking whether this case would be possible, but for completion makes sense to add it I guess

Reply:
Yes, it is more for the sake of completeness.